Source/DafnyLanguageServer/Workspace/VerificationProgressReporter.cs

```diff
@@ -122,18 +124,31 @@ IDiagnosticPublisher diagnosticPublisher
   /// Also set the implementation priority depending on the last edited methods 
   /// </summary>
   /// <param name="implementations">The implementations to be verified</param>
-  public void ReportImplementationsBeforeVerification(Implementation[] implementations) {
+  public virtual void ReportImplementationsBeforeVerification(Implementation[] implementations) {
     if (document.LoadCanceled || implementations.Length == 0) {
       return;
     }
+    var newLastTouchedMethodPositions = document.LastTouchedMethodPositions.ToList();
+    var newlyTouchedVerificationTree = document.VerificationTree.Children.FirstOrDefault(node =>
+      node != null && document.LastChange != null && node.Range.Contains(document.LastChange), null);
+    if (newlyTouchedVerificationTree != null) {
+      RememberLastTouchedMethodPositions(newlyTouchedVerificationTree.Position, newLastTouchedMethodPositions);
+      document.LastTouchedMethodPositions = newLastTouchedMethodPositions.TakeLast(5).ToImmutableList();
```

Comment:
Wouldn't it be easier to determine last changed locations in `ApplyChangesAsync`, since that's where you have the `DidChangeTextDocumentParams documentChange` ?